stko/optimizers/collapser.py

```diff
@@ -228,60 +231,571 @@ def optimize(self, mol):
+
+    def _non_bond_potential(self, distance):
+        """
+        Define an arbitrary repulsive nonbonded potential.
+
+        This potential has no relation to an empircal forcefield.
+
+        """
+
+        return (
+            self._nonbond_epsilon * (
+                (self._nonbond_sigma/distance) ** self._nonbond_mu
+            )
+        )
+
+    def _compute_non_bonded_potential(self, mol):
+
+        # Get all pairwise distances.
+        pair_dists = pdist(mol.get_position_matrix())
+        non_bonded_potential = self._non_bond_potential(pair_dists)
+
+        return sum(non_bonded_potential)
+
+    def _compute_potential(self, mol, long_bond_infos):
+
+        system_potential = self._compute_non_bonded_potential(mol)
+        for long_bond_ids in long_bond_infos:
+            long_bond = long_bond_infos[long_bond_ids]
+
+            system_potential += self._bond_potential(
+                distance=self._get_bond_length(
+                    mol=mol,
+                    bond=long_bond.get_bond()
+                )
+            )
+
+        return system_potential
+
+    def _translate_atoms_along_vector(self, mol, atom_ids, vector):
+
+        new_position_matrix = mol.get_position_matrix()
+        for atom in mol.get_atom_infos(atom_ids=atom_ids):
+            id = atom.get_atom().get_id()
+            pos = mol.get_position_matrix()[id]
+            new_position_matrix[id] = pos - vector
+
+        return mol.with_position_matrix(new_position_matrix)
+
+    def _rotate_atoms_onto_vector(
+        self,
+        mol,
+        atom_ids,
+        start_vector,
+        target_vector,
+        axis
+    ):
+
+        return None
+
+    def _test_move(self, curr_pot, new_pot):
+
+        if new_pot < curr_pot:
+            return True
+        else:
+            exp_term = np.exp(-self._beta*(new_pot-curr_pot))
+            rand_number = random.random()
+
+            if exp_term > rand_number:
+                return True
+            else:
+                return False
+
+    def _output_top_lines(self):
+
+        string = (
+            '====================================================\n'
+            '                Collapser optimisation              \n'
+            '                ----------------------              \n'
+            '                                                    \n'
+            f' step size = {self._step_size} \n'
+            f' target bond length = {self._target_bond_length} \n'
+            f' num. steps = {self._num_steps} \n'
+            f' bond epsilon = {self._bond_epsilon} \n'
+            f' nonbond epsilon = {self._nonbond_epsilon} \n'
+            f' nonbond sigma = {self._nonbond_sigma} \n'
+            f' nonbond mu = {self._nonbond_mu} \n'
+            f' beta = {self._beta} \n'
+            '====================================================\n\n'
+        )
+
+        return string
+
+    def _plot_progess(self, steps, maxds, spots, npots, output_dir):
```

Comment:
I'm kind og against adding plotting code for now. I think there are too many options the user may want when plotting. Also, there's not really an customisability as the plots are not returned.

Reply:
Okay happy to leave in after reading Slack message